logic/src/commonMain/kotlin/com/wire/kalium/logic/data/wireuser/WireUserRepository.kt

```diff
@@ -12,27 +11,37 @@ import com.wire.kalium.network.api.contact.search.ContactSearchRequest
 import com.wire.kalium.network.api.user.details.ListUserRequest
 import com.wire.kalium.network.api.user.details.UserDetailsApi
 import com.wire.kalium.network.api.user.details.qualifiedIds
+import com.wire.kalium.persistence.dao.UserDAO
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 
-
-interface PublicUserRepository {
+interface WireUserRepository {
```

Comment:
I think we need reconsider the naming a bit. We now have `UserRespository` and `WireUserRepository`, in my head this refers to the same thing since all users we are talking about are "wire" users.

My understanding was that you wanted to extract the search functionality from the UserRepository. Wouldn't `SearchUserRepository` and `SearchUser` be a fitting name, but I'm open for suggestions.

Reply:
yes, that could be an option as well to create a SearchUserRepository, however I think the confusion is because of the UserRepository being called UserRepository which it should be called SelfUserRepository instead I think, then this repository would work with other users than the self user, this PR is approved by alexandre already so let me know what you think of it. I do not want to bring confusion into the Kalium :P

Reply:
Well the confusing is already happening :)

`UserRepository` today is not only about the self user it also enables you to access any of the users you are connected to.

But your concern right is about searching for users so maybe we should just focus on implementing that once we have more use cases involving users it will be more clear what's the best way is to split the repositories.

Reply:
true, now I read the repo there are some methods that are not acting on the SelfUser maybe it is a sign to split those methods though instead of keeping ti inside the UserRepository , dunno I get the felling that UserRepository gets kinds of two responsibilities :P 

so what is your suggestion, should we merge it the way it is now in my PR and keep in mind that we could change it later on or should we discuss it more internally